### ISSUE TITLE
Incorrect `unused_import` lint firing for a function-local `macro_rules!`

### ISSUE THREAD

**Issue by danielhenrymantilla** [rust-lang/rust#78894]
I tried this code:

```rust
fn main ()
{
    m!();
    // where
    macro_rules! m {
        () => ();
    } use m;
}
```

I expected to see this happen: no warning.

Instead, this happened: [I get a warning](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=21509e89ac39afe8028e563cbf8530c6) saying:

```text
warning: the item `m` is imported redundantly
 --> src/main.rs:7:11
  |
5 | /     macro_rules! m {
6 | |         () => ();
7 | |     } use m;
  | |     -     ^
  | |_____|
  |       the item `m` is already defined here
  |
  = note: `#[warn(unused_imports)]` on by default
```

This is incorrect since removing that `use` "statement" leads to the code not compiling.

### Meta

Affects all three versions currently available in the Playground (1.47.0 -> 1.49.0)

**Comment by petrochenkov** [comment]
This issue no longer reproduces and can be closed when a corresponding test is added.
